src/components/UIFocusText.cpp

```diff
@@ -0,0 +1,110 @@
+#include "UIFocusText.hpp"
+#include <GUI/BsGUILabel.h>
+#include <GUI/BsGUIPanel.h>
+#include <RTTI/RTTI_UIFocusText.hpp>
+#include <Renderer/BsCamera.h>
+#include <components/Focusable.hpp>
+
+namespace REGoth
+{
+  UIFocusText::UIFocusText(const bs::HSceneObject& parent, HUIElement parentUiElement)
+      : UIElement(parent, parentUiElement, new bs::GUIPanel())
+  {
+    setName("UIFocusText");
+
+    mLabelAboveObject = layout().addNewElement<bs::GUILabel>(bs::HString(""));
+    // mLabelAboveObject->setStyle("GothicLabelObjectFocus");
+  }
+
+  UIFocusText::~UIFocusText()
+  {
+  }
+
+  void UIFocusText::putTextAbove(HFocusable focusable, bs::String text)
```

Comment:
Could you go into more details on how this will work? The way I see it right now there is only one UIFocusText component that handels all the Focus draws (which makes sense since there can only be one thing in front of you to focus). 
But when would this called and why would we provide a text argument instead of using getText on the Focusable?

Reply:
Good catch. The `UIFocusText` actually existed before the `Focusable`-Component. "Back then" the method wanted a `bs::HSceneObject` and the text to display.

I later figured out I should add a component to the focusable stuff.